frontend/middleware_validatesubscription.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/api/subscription"
+)
+
+type SubscriptionStateMuxValidator struct {
+	cache *Cache
+}
+
+func NewSubscriptionStateMuxValidator(c *Cache) *SubscriptionStateMuxValidator {
+	return &SubscriptionStateMuxValidator{
+		cache: c,
+	}
+}
+
+func (s *SubscriptionStateMuxValidator) MiddlewareValidateSubscriptionState(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
```

Comment:
We could add a "return" in each of the if/else branches of the functions. We would convert the else's to simple ifs and avoid nesting the levels. WDT? 🙂

Reply:
yeah, this logic is a mess, just a rough outline of what needs to get checked. Returning early is something I'll add in :)